ref/Microsoft.Build/net/Microsoft.Build.cs

```diff
@@ -872,6 +872,7 @@ public partial class EvaluationContext
     {
         internal EvaluationContext() { }
         public static Microsoft.Build.Evaluation.Context.EvaluationContext Create(Microsoft.Build.Evaluation.Context.EvaluationContext.SharingPolicy policy) { throw null; }
+        public static Microsoft.Build.Evaluation.Context.EvaluationContext Create(Microsoft.Build.Evaluation.Context.EvaluationContext.SharingPolicy policy, Microsoft.Build.FileSystem.IDirectoryCacheFactory directoryCacheFactory) { throw null; }
```

Comment:
A problem of this design is that we need change every code calls into RevaluateProjectIfNecessary, which can be very difficult. Any chance we can provide the CacheFactory to the project, which is used in the lifetime of the project?  If the evaluation should not use cache, maybe it can be controlled on CPS side?

Reply:
If there are no scenarios where you would want to evaluate the same `Project` with different backing caches, then I guess we could add `IDirectoryCache` as a `Project` constructor parameter. It's not as clean, though, because this conceptually belongs in evaluation context. Which I thought CPS wanted to use for (all?) evaluations anyway.